### PR TITLE
Monthly Release 2021.07.0 Developer Documentation Updates

### PR DESCRIPTION
**Pull request by brianduffield-okta**
## Description:
- **What's changed?** Adding monthly release notes and applicable Developer Doc updates.
- **Is this PR related to a Monolith release?** 2021.07.0

### Resolves:

* [OKTA-408752](https://oktainc.atlassian.net/browse/OKTA-408752)

### CODE REVIEW

File: packages/@okta/vuepress-site/docs/reference/api/schemas/index.md [hunk 1]
```diff
@@ -2462,3 +2464,528 @@ Specific property types support a subset of [JSON Schema validations](https://to
...
 | `integer`     | [JSON Number](https://tools.ietf.org/html/rfc7159#section-6) with 32-bit signed two's complement integer constraint           | `minimum` and `maximum`     |
 | `array`       | [JSON Array](https://tools.ietf.org/html/rfc7159#section-5)                                                                         |                             |
+
+## Unique attributes
+
+You can enforce uniqueness for custom properties in Okta user profiles or the Okta group profile, such as an employee identification number. You can declare a maximum of five unique properties for each user type and five unique properties in the Okta group profile. Different user types can have the same or different unique properties (up to the limit of five per type).
+
+Unique properties in Okta user profiles share a single namespace across all [user types](/docs/reference/api/user-types) in an org. If user types A and B both contain the property `ice cream` and you identify it as unique in both profiles, then if a user of type A has the value `chocolate`, no other users of type A or B (or any other user type which declares `ice cream` as unique) can have that value.
```
Comment (susanharper-okta) [inline packages/@okta/vuepress-site/docs/reference/api/schemas/index.md:2472]:
```suggestion
Unique properties in Okta user profiles share a single namespace across all [user types](/docs/reference/api/user-types) in an org. If user types A and B both contain the property `ice cream` and you identify it as unique in both profiles, then if a user of type A has the value `chocolate`, no other users of type A or B (or any other user type that declares `ice cream` as unique) can have that value.
```

File: packages/@okta/vuepress-site/docs/reference/api/schemas/index.md [hunk 2]
```diff
@@ -2462,3 +2464,528 @@ Specific property types support a subset of [JSON Schema validations](https://to
...
+
+Properties that aren't unique also aren't tracked for uniqueness. Suppose the property `candy` is unique in type E and not unique in type F. If a user of type E has the value `caramel` for the `candy` property, no other users of type E can also have the value `caramel`, but any number of users of type F can already have or later be assigned the value `caramel`. Furthermore, because `candy` is not unique in type F, any values users of type F may have are not considered when enforcing uniqueness for users of type E. No matter how many users of type F already have the value `cotton`, it may be assigned to a user of type E as long as no other such user already has that value.
+
+If you attempt to create or update a user with a duplicate value for a custom user property with a uniqueness restriction, the user creation or update operation fails. The user isn't created or updated until you enter a unique value. Similarly, creating or updating a group will fail when the request contains a value for a unique custom group property that is duplicated by another group.
```
Comment (susanharper-okta) [inline packages/@okta/vuepress-site/docs/reference/api/schemas/index.md:2476]:
```suggestion
If you attempt to create or update a user with a duplicate value for a custom user property with a uniqueness restriction, the user creation or update operation fails. The user isn't created or updated until you enter a unique value. Similarly, creating or updating a group fails when the request contains a value for a unique custom group property that is duplicated by another group.
```

File: packages/@okta/vuepress-site/docs/reference/api/schemas/index.md [hunk 3]
```diff
@@ -2462,3 +2464,528 @@ Specific property types support a subset of [JSON Schema validations](https://to
...
+
+`null` values don't enter into the uniqueness calculation. If the unique property is not also specified as being required, you can also omit the value entirely; multiple users or groups can omit the property and not violate uniqueness.
```
Comment (susanharper-okta) [inline packages/@okta/vuepress-site/docs/reference/api/schemas/index.md:2478]:
```suggestion
`null` values don't enter into the uniqueness calculation. If the unique property isn't also specified as being required, you can also omit the value entirely. Multiple users or groups can omit the property and not violate uniqueness.
```

File: packages/@okta/vuepress-site/docs/reference/api/schemas/index.md [hunk 4]
```diff
@@ -2462,3 +2464,528 @@ Specific property types support a subset of [JSON Schema validations](https://to
...
+
+After the request to mark existing custom properties as unique is submitted, an asynchronous validation check is performed to make sure that there are no existing duplicate entries. If you have a significant number of users or groups, the validation can take some time.
+
+A uniqueness status of `"unique": "PENDING_UNIQUENESS"` indicates that the validation check is still in progress. Use the Universal Directory page in the Admin Console (**Directory** > **Directory Integrations**) to track the status of the validation check. After the validation completes, if you submit a [get user schema](#get-user-schema) request or a [get group schema](#get-group-schema) request, the property's uniqueness status changes to `UNIQUE_VALIDATED` if no duplicate records are found, and uniqueness is then enforced on that property. Otherwise, if duplicate records are found, the `unique` attribute of the schema property isn't shown in the get schema request and uniqueness isn't enforced on the schema property.
+
+##### Request example
+
+For the request example below, assume that the default user profile schema contains the custom property `twitterUserName` that is not unique.
```
Comment (susanharper-okta) [inline packages/@okta/vuepress-site/docs/reference/api/schemas/index.md:2646]:
```suggestion
For the following request example, assume that the default user profile schema contains the custom property `twitterUserName` that isn't unique.
```

File: packages/@okta/vuepress-site/docs/reference/api/schemas/index.md [hunk 5]
```diff
@@ -2462,3 +2464,528 @@ Specific property types support a subset of [JSON Schema validations](https://to
...
+        }
+    }
+}
+```
+
+### Update existing unique custom property to be non-unique
+
+You can use the [update user profile schema property](#update-user-profile-schema-property) request or the [update group profile schema property](#update-group-profile-schema-property) request to change existing unique custom user or group properties to be non-unique by specifying `"unique": false` on the properties to be changed to non-unique. The response shows the properties without the `unique` attribute and the uniqueness constraint is then removed on those properties.
+
+Note that if multiple user types declare a property as unique and you remove the uniqueness constraint on one type, there may be a delay before users of other types which declare the property as unique can be assigned values formerly held by users of the first type.
```
Comment (susanharper-okta) [inline packages/@okta/vuepress-site/docs/reference/api/schemas/index.md:2820]:
```suggestion
Note that if multiple user types declare a property as unique and you remove the uniqueness constraint on one type, there may be a delay before users of other types that declare the property as unique can be assigned values formerly held by users of the first type.
```

File: packages/@okta/vuepress-site/docs/reference/api/schemas/index.md [hunk 6]
```diff
@@ -2462,3 +2464,528 @@ Specific property types support a subset of [JSON Schema validations](https://to
...
+
+##### Request example
+
+For the request example below, assume that the default user profile schema contains the custom property `twitterUserName` that is unique.
```
Comment (susanharper-okta) [inline packages/@okta/vuepress-site/docs/reference/api/schemas/index.md:2824]:
```suggestion
For the following request example, assume that the default user profile schema contains the custom property `twitterUserName` that is unique.
```

File: packages/@okta/vuepress-site/docs/release-notes/2021/index.md [hunk 7]
```diff
@@ -1,6 +1,50 @@
...
+
+#### Rate limit events for user and session user activity
+
+Previously, rate limit violation events for user and session user were logged as org-wide system events (`system.org.rate_limit.violation`). These rate limit violation events are now logged more specifically as [operation events](/docs/reference/rl-system-log-events/#operation-rate-limits) (`system.operation.rate_limit.violation`). <!--OKTA-394607-->
+
+#### Schema API feature unique attributes is now GA in Preview
+
+The Schemas API now includes [unique attributes](/docs/reference/api/schemas/#uniqueattributes) for custom properties in Okta user profiles and the Okta Group profile. You can declare a maximum of five unique properties for each user type and five unique properties in the Okta group profile. This feature helps prevent the duplication of data and ensures data integrity. <!--OKTA-400824-->
```
Comment (susanharper-okta) [inline packages/@okta/vuepress-site/docs/release-notes/2021/index.md:38]: is it Group or group? User or user?

File: packages/@okta/vuepress-site/docs/release-notes/2021/index.md [hunk 8]
```diff
@@ -1,6 +1,50 @@
...
+
+#### Bugs fixed in 2021.07.0
+
+- The IdP claim was not available in the `id_token` or included with the Token Inline Hook request. (OKTA-407459)
```
Comment (susanharper-okta) [inline packages/@okta/vuepress-site/docs/release-notes/2021/index.md:42]:
```suggestion
- The IdP claim wasn't available in the `id_token` or included with the Token Inline Hook request. (OKTA-407459)
```

File: packages/@okta/vuepress-site/docs/release-notes/2021/index.md [hunk 9]
```diff
@@ -1,6 +1,50 @@
...
+
+- When the Users lifecycle API `users/{{userId}}/lifecycle/reset_factors` was called to reset user factors, a status 403 error was received, even with a valid bearer token and scope (`okta.users.manage`). (OKTA-404613)
+
+- When an OIDC client app was created, the [Apps API](/docs/reference/api/apps) call could not modify the `visibility.hide` property. (OKTA-399408)
```
Comment (susanharper-okta) [inline packages/@okta/vuepress-site/docs/release-notes/2021/index.md:46]:
```suggestion
- When an OIDC client app was created, the [Apps API](/docs/reference/api/apps) call couldn't modify the `visibility.hide` property. (OKTA-399408)
```

**Review by susanharper-okta — COMMENTED** [review]
couple of things I should have caught when reviewing that PR earlier...and some style nits....question - why is the navbar changes in your PR?

File: packages/@okta/vuepress-site/docs/release-notes/2021/index.md (hunk 7 above)
Comment (brianduffield-okta) [reply packages/@okta/vuepress-site/docs/release-notes/2021/index.md:38]: Appears to be capitalized. Will update

**Review by susanharper-okta — APPROVED** [review]
🚢 lgtm

**Pull request merged**